File: Creational/Abstract Factory/Python/ConcreteFactory1.py
from AbstractFactory import *
from ConcreteProductA_1 import *
from ConcreteProductB_1 import *
from ConcreteProductC_1 import *

class ConcreteFactory1(AbstractFactory):
	
	def create_productA(self):
		return ConcreteProductA_1()
		
	def create_productB(self):
		return ConcreteProductB_1()
		
	def create_productC(self):
		return ConcreteProductC_1()

File: Creational/Abstract Factory/Golang/ConcreteFactory1.go
package main

type ConcreteFactory1 struct{}

func NewConcreteFactory1() AbstractFactory {
	return &ConcreteFactory1{}
}

func (*ConcreteFactory1) CreateProductA() AbstractProductA {
	return &ConcreteProductA_1{}
}

func (*ConcreteFactory1) CreateProductB() AbstractProductB {
	return &ConcreteProductB_1{}
}

func (*ConcreteFactory1) CreateProductC() AbstractProductC {
	return &ConcreteProductC_1{}
}
